### PR TITLE
fix: skip highlighter on empty search results

### PR DESCRIPTION
**Pull request by junjiejiangjjj**
issue: #50381

  Search results can contain non-empty FieldsData even when there are no
  matched rows, because FieldsData may only hold empty output-field templates.
  Use result IDs to detect zero-hit search results before running lexical or
  semantic highlight processing.

### CODE REVIEW

File: internal/proxy/highlighter.go [hunk 1]
```diff
@@ -322,10 +322,25 @@ func newLexicalHighlightOperator(t *searchTask, tasks []*highlightTask) (operato
...
+}
+
 func (op *lexicalHighlightOperator) run(ctx context.Context, span trace.Span, inputs ...any) ([]any, error) {
 	result := inputs[0].(*milvuspb.SearchResults)
-	datas := result.GetResults().GetFieldsData()
-	// skip highlight if result is empty
+	resultData := result.GetResults()
+	// skip highlight if result has no hits. FieldsData may contain empty field templates
+	// even when Topks/Ids/Scores show zero matched rows.
+	if !hasSearchResultHits(resultData) {
```
Comment (liliu-z) [inline internal/proxy/highlighter.go:339]: The new guard `if !hasSearchResultHits(resultData) { return []any{result}, nil }` short-circuits before any field lookup, but 7 existing SearchPipelineSuite tests use fixtures that never populate `Ids`: TestHighlightOp fails on an unmet mockery expectation at cleanup, TestSemanticHighlightOpMissingField loses its expected error, and five assertion-based semantic tests now see nil `HighlightResults`, so CI fails on every run since the diff contains no test updates. Update the highlight fixtures to populate `Ids`, and add the regression test the issue explicitly requested — zero hits with non-empty zero-row `FieldsData` templates passing through both the lexical and semantic `run()` operators.

File: internal/proxy/highlighter.go [hunk 2]
```diff
@@ -464,7 +479,12 @@ type semanticHighlightOperator struct {
 
 func (op *semanticHighlightOperator) run(ctx context.Context, span trace.Span, inputs ...any) ([]any, error) {
 	result := inputs[0].(*milvuspb.SearchResults)
-	datas := result.Results.GetFieldsData()
+	resultData := result.GetResults()
+	if !hasSearchResultHits(resultData) {
```
Comment (liliu-z) [inline internal/proxy/highlighter.go:483]: This guard silently flips the zero-hit semantic highlight contract pinned by TestSemanticHighlightOpEmptyResults (added with the semantic highlight feature in #46189): a zero-hit response previously carried one HighlightResult per field with empty Datas, and now carries no HighlightResults at all — a wire-visible change for clients. The change is almost certainly the intended fix (it matches the pre-#48984 shape, before the Go-based reduce started emitting non-empty zero-row FieldsData templates), but unlike the other six broken fixtures this test cannot be repaired by populating Ids — it exists specifically to pin the zero-hit path — so its expectations must be flipped to record the new contract explicitly rather than leaving the change as an untested side effect.

File: internal/proxy/highlighter.go (hunk 1 above)
Comment (junjiejiangjjj) [reply internal/proxy/highlighter.go:339]: Thanks, fixed. I populated IDs in the non-zero-hit highlight fixtures and added regression coverage for lexical/semantic zero-hit with non-empty FieldsData. Also added coverage for non-zero-hit with empty FieldsData, which now returns an explicit error.

File: internal/proxy/highlighter.go (hunk 2 above)
Comment (junjiejiangjjj) [reply internal/proxy/highlighter.go:483]: Agreed, updated. TestSemanticHighlightOpEmptyResults now explicitly records the new zero-hit contract: no result IDs means highlight is skipped and HighlightResults stays empty.
